### PR TITLE
migrate to winapi 0.3

### PR DESCRIPTION
**Pull request by steffengy**
- [x] Some of those changes should also be applied to cargo, to keep those common files consistent.
filed https://github.com/rust-lang/cargo/pull/4877

### CODE REVIEW

**Comment by steffengy** [comment]
@alexcrichton rebased

**Comment by steffengy** [comment]
@alexcrichton 
Well, seems like I should've checked with `cargo check --all-features`.  
`get_msi_product_code` now also compiles.

File: src/rustup-win-installer/src/lib.rs [hunk 1]
```diff
@@ -6,7 +6,9 @@ extern crate rustup;
 use std::ffi::CString;
 use std::path::PathBuf;
 use std::collections::HashMap;
-use ::winapi::{HRESULT, PCSTR, UINT, LPCWSTR, LPWSTR, LPVOID};
+
+use winapi::shared::ntdef::{HRESULT, PCSTR, LPCWSTR, LPWSTR, ;
```
Comment (retep998) [inline src/rustup-win-installer/src/lib.rs]: `, ;`

**Comment by steffengy** [comment]
@alexcrichton 
Or rather with `cargo check --all-features --all`. And another hopefully final round, please.

**Pull request merged**